### PR TITLE
Automation of the BZ#2053021-Ceph conmon [DBG] logs flood /var/log/messages causing filesystem to fill up

### PR DESCRIPTION
**Pull request by SrinivasaBharath**
# Description

Automation of the customer issue BZ#2053021
The steps are updated at - CEPH-83575403
Automation scripts -
  1.  Create a pool            
  2. Push the data into the pool
  3. Identify any PG id to initiate the scrub.
  4. Perform the scrub on any PG
  5. Check the ceph logs/journalID messages for the following lines-     
                         <pg-id> scrub starts      
                          <pg-id> scrub ok    
 6. Change the "mon_cluster_log_level"  parameter value to warn  
 7.   Check the ceph logs/journalID messages for the following lines and lines should not get generate in the ceph/journalctl logs-     
                         <pg-id> scrub starts      
                          <pg-id> scrub ok   
8. Execute the same steps by performing the deep scrub.       
Jira tas - https://issues.redhat.com/browse/RHCEPHQE-15809
Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by SrinivasaBharath** [comment]
Quincy pass log - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-H57XV7

File: tests/rados/test_scrub_log.py [hunk 1]
```diff
@@ -17,94 +18,221 @@
...
+        log.error(
+            f"The mon_cluster_log_file_level parameter not deleted and the value is {parm_out_put_check}"
+        )
+        return 1
+    try:
+        parm_out_put_check = mon_obj.get_config(
+            section="mon", param="mon_cluster_log_to_syslog_level"
+        )
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:62]: Update the error message, And also check the error string seen on cluster.

File: tests/rados/test_scrub_log.py [hunk 2]
```diff
@@ -17,94 +18,221 @@
...
+    if parm_out_put_check is not None:
+        log.error(
+            f"The mon_cluster_log_to_syslog_level parameter not deleted and the value is {parm_out_put_check}"
+        )
+        return 1
+    parm_out_put_check = mon_obj.get_config(
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:68]: Add this also in a try catch block.

File: tests/rados/test_scrub_log.py [hunk 3]
```diff
@@ -17,94 +18,221 @@
 
 def run(ceph_cluster, **kw) -> int:
     """
-    Test to verify if adequate logging is generated in the OSD logs upon initiating scrub & deep-scrubs
+    1.Test to verify if adequate logging is generated in the OSD logs upon initiating scrub & deep-scrubs
+    2. Verification of the logs if the mon_cluster_log_level values are warn and debug.More details are
+       updated at - https://bugzilla.redhat.com/show_bug.cgi?id=2053021
     Returns:
         1 -> Fail, 0 -> Pass
     """
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:26]: Make the bug verification optional via a flag passed from suite file

File: tests/rados/test_scrub_log.py (hunk 2 above)
Comment (SrinivasaBharath) [reply tests/rados/test_scrub_log.py]: Included the try-catch block

File: tests/rados/test_scrub_log.py (hunk 3 above)
Comment (SrinivasaBharath) [reply tests/rados/test_scrub_log.py:26]: Changed the logic and provide the flag.

**Comment by SrinivasaBharath** [comment]
**Logs:**

Squid -
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7RPY1C- Pass with the verify_log  flag false
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-HZNXLK - Fail with the verify_log flag true

Reef-
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-BUX2I3/ -  Pass with the verify_log  flag true
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-NVOSGO - Pass with the verify_log  flag false 


Quincy-
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-UFK0LF - Pass with the verify_log  flag flag true
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WXS9WT - Pass log with the verify_log flag false

Pacific-
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ABHP8G -  Pass log with the verify_log flag false

File: tests/rados/test_scrub_log.py [hunk 4]
```diff
@@ -17,94 +18,238 @@
...
         return 0
+    parm_out_put_check = None
+    # Check that paramters are removed from the build
+    if verify_log_check:
+        try:
+            parm_out_put_check = mon_obj.get_config(
+                section="mon", param="mon_cluster_log_file_level"
+            )
+        except Exception as e:
+            log.error(f"An unexpected error occurred: {e}")
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:53]: should there be a return statement here post failure?

File: tests/rados/test_scrub_log.py [hunk 5]
```diff
@@ -17,94 +18,238 @@
...
+            log.error(
+                f"The mon_cluster_log_file_level parameter not deleted and the value is {parm_out_put_check}"
+            )
+            return 1
+        try:
+            parm_out_put_check = mon_obj.get_config(
+                section="mon", param="mon_cluster_log_to_syslog_level"
+            )
+        except Exception as e:
+            log.error(f"An unexpected error occurred: {e}")
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:64]: should there be a return statement here post failure?

File: tests/rados/test_scrub_log.py [hunk 6]
```diff
@@ -17,94 +18,238 @@
...
+            log.error(
+                f"The mon_cluster_log_to_syslog_level parameter not deleted and the value is {parm_out_put_check}"
+            )
+            return 1
+        try:
+            parm_out_put_check = mon_obj.get_config(
+                section="mon", param="mon_cluster_log_level"
+            )
+        except Exception as e:
+            log.error(f"An unexpected error occurred: {e}")
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:75]: should there be a return statement here post failure? Or else, we need to update the logging level and improve the error message here.

File: tests/rados/test_scrub_log.py [hunk 7]
```diff
@@ -17,94 +18,238 @@
...
-                pool_name=pool["pool_name"],
-                max_objs=50,
-                check_ec=False,
-                rados_write_duration=20,
+        for log_level in ["debug", "warn"]:
+            log.info(
+                f"====START:Testing the scenario with the mon_cluster_log_level value {log_level} ======="
+            )
+            # Get the ceph log size before starting the test
+            before_test_log_size = get_file_size(rados_obj, installer)
```
Comment (pdhiran) [inline tests/rados/test_scrub_log.py:90]: I believe that the logs like "ceph.log" whose size is checked here, is only generated when "mon_cluster_log_to_file" param is true. Can you confirm if that is indeed the case? 

If Yes, Please make sure to enable the param on the cluster before starting the tests.

Note : As part of def log_to_file() , we are enabling the param, but incase that method is not called during cluster set-up, the logs might not be present on the cluster. better to make sure.

**Comment by pdhiran** [comment]
The log files are too huge, and hence are not captured from beginning to end. Let's find some way to reduce the log lines generated.

File: tests/rados/test_scrub_log.py (hunk 4 above)
Comment (SrinivasaBharath) [reply tests/rados/test_scrub_log.py:53]: Added the return statement.

File: tests/rados/test_scrub_log.py (hunk 5 above)
Comment (SrinivasaBharath) [reply tests/rados/test_scrub_log.py:64]: Added the return statement.

File: tests/rados/test_scrub_log.py (hunk 6 above)
Comment (SrinivasaBharath) [reply tests/rados/test_scrub_log.py:75]: I am verifying the output in the next lines.

**Comment by SrinivasaBharath** [comment]
> The log files are too huge, and hence are not captured from beginning to end. Let's find some way to reduce the log lines generated.

The logs are huge because I  executed the script with the --log-level as DEBUG mode. 
Please find the below log with the INFO mode-
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-CZ4THI/

**Comment by SrinivasaBharath** [comment]
Pass log- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-A0AH37

**Comment by SrinivasaBharath** [comment]
Modified the code and tested  without adding the following test in the test suite-

- test:
      name: Enable logging to file
      module: rados_prep.py
      config:
        log_to_file: true
      desc: Change config options to enable logging to file

Pass log- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-P659BA

File: tests/rados/test_scrub_log.py (hunk 7 above)
Comment (SrinivasaBharath) [reply tests/rados/test_scrub_log.py:98]: Included the enable_file_logging method in the code.

**Review by pdhiran — APPROVED** [review]
Looks Good.

**Pull request merged**